### PR TITLE
Add tests and fix bug in byTracking plugin

### PR DESCRIPTION
**Pull request by MadeByMike**
Add tests and fix bug in tracking plugin.

### CODE REVIEW

**Comment by MadeByMike** [comment]
See my comment about the AuthRelationship field - I think we should remove it from here. It will make testing easier and will remove complexity. we rely on it ONLY to set the value since we provide custom access control. Sadly it's only in the way here and I regret that we added it.

**Comment by MadeByMike** [comment]
Closes #3332

File: packages/list-plugins/lib/tracking/byTracking.js [hunk 1]
```diff
@@ -25,9 +25,12 @@ const _byTracking = ({ created = true, updated = true }) => ({
   }
 
   const newResolveInput = ({ resolvedData, operation, originalInput, context }) => {
+    // if no data received from the mutation, skip the update
+    if (Object.keys(originalInput).length === 0) {
+      return resolvedData;
+    }
+
```
Comment (MadeByMike) [inline packages/list-plugins/lib/tracking/byTracking.js:32]:
```suggestion
```

File: packages/list-plugins/lib/tracking/byTracking.js [hunk 2]
```diff
@@ -26,8 +26,6 @@ const _byTracking = ({ created = true, updated = true }) => ({
 
   const newResolveInput = ({ resolvedData, operation, originalInput, context }) => {
```
Comment (timleslie) [inline packages/list-plugins/lib/tracking/byTracking.js:27]:
```suggestion
  const newResolveInput = ({ resolvedData, operation, context }) => {
```

**Comment by MadeByMike** [comment]
Removed the following from atTracking for consistency:

```
      // if no data received from the mutation, skip the update
      if (Object.keys(originalInput).length === 0) {
        return resolvedData;
      }
```

To document reasoning. It's valid that a create or update may cause a side effect that system admins want to track. This is not about "has the data been input into the form?" as much as "has the create/update been triggered, regardless of changes?"

**Comment by gautamsi** [comment]
that makes sense, we can live with that for now and in later PR we can make this configurable.

**Pull request merged**